src/mds/MDCache.cc

```diff
@@ -7780,12 +7779,10 @@ void MDCache::trim_client_leases()
   }
 }
 
-void MDCache::check_memory_usage()
+void MDCache::check_memory_usage(const MemoryModel::snap &baseline)
 {
-  static MemoryModel mm(g_ceph_context);
-  static MemoryModel::snap last;
-  mm.sample(&last);
-  static MemoryModel::snap baseline = last;
```

Comment:
since the code to get the baseline has been shifted in upkeep thread, does this mean till now we had wrong values of baseline and this basically means `baseline.get_heap()` was same as `last.get_heap()`?!

Reply:
Not really. Looks similar to me.